### PR TITLE
Interfaced handlers for RefreshView and SearchBar

### PR DESCRIPTION
**Pull request by antonfirsov**
Contributes to https://github.com/dotnet/maui/issues/4378.

### CODE REVIEW

File: src/Core/src/Handlers/SearchBar/ISearchBarHandler.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+using QueryEditor = System.Object;
+#endif
+
+namespace Microsoft.Maui.Handlers
+{
+	public partial interface ISearchBarHandler : IViewHandler
+	{
+		new ISearchBar VirtualView { get; }
+		new PlatformView PlatformView { get; }
+		QueryEditor? QueryEditor { get; }
```
Comment (antonfirsov) [inline src/Core/src/Handlers/SearchBar/ISearchBarHandler.cs:22]: This seemed to be a straightforward addition greatly simplifying handler implementations.

**Review by rmarinho — CHANGES_REQUESTED** [review]
```
/Users/builder/azdo/_work/1/s/src/Controls/src/Core/HandlerImpl/SearchBar/SearchBar.cs(15,21): error CS0117: 'SearchBarHandler' does not contain a definition for 'SearchBarMapper' [/Users/builder/azdo/_work/1/s/src/Controls/src/Core/Controls.Core.csproj]
```

File: src/Core/src/Handlers/SearchBar/SearchBarHandler.Android.cs [hunk 2]
```diff
@@ -44,69 +44,69 @@ void SetupDefaults(SearchView platformView)
...
 		{
 			handler.PlatformView?.UpdatePlaceholder(searchBar);
 		}
 
-		public static void MapPlaceholderColor(SearchBarHandler handler, ISearchBar searchBar)
+		public static void MapPlaceholderColor(ISearchBarHandler handler, ISearchBar searchBar)
 		{
-			handler.PlatformView?.UpdatePlaceholderColor(searchBar, DefaultPlaceholderTextColors, handler._editText);
+			if (handler is SearchBarHandler platformHandler)
+				platformHandler.PlatformView?.UpdatePlaceholderColor(searchBar, DefaultPlaceholderTextColors, platformHandler._editText);
```
Comment (PureWeen) [inline src/Core/src/Handlers/SearchBar/SearchBarHandler.Android.cs:65]: Can this `_editText` be changed to `QueryEditor`?

File: src/Core/src/Handlers/SearchBar/SearchBarHandler.Windows.cs [hunk 3]
```diff
@@ -39,66 +41,87 @@ protected override void DisconnectHandler(AutoSuggestBox platformView)
...
+		public static void MapTextColor(ISearchBarHandler handler, ISearchBar searchBar)
 		{
-			handler.PlatformView?.UpdateTextColor(searchBar, handler._defaultTextColorBrush, handler._defaultTextColorFocusBrush, handler._queryTextBox);
+			if (handler is SearchBarHandler platformHandler)
+			{
+				handler.PlatformView?.UpdateTextColor(
+					searchBar,
+					platformHandler._defaultTextColorBrush,
+					platformHandler._defaultTextColorFocusBrush,
+					platformHandler._queryTextBox);
```
Comment (PureWeen) [inline src/Core/src/Handlers/SearchBar/SearchBarHandler.Windows.cs:96]: `QueryEditor` ?

**Review by PureWeen — CHANGES_REQUESTED** [review]

**Review by PureWeen — APPROVED** [review]

**Pull request merged**